app/src/main/java/hr/foi/air/servicesync/ui/screens/ServiceReservationScreen.kt

```diff
@@ -70,41 +75,30 @@ fun ServiceReservationScreen(serviceName: String, companyId: String) {
         )
 
         if (showDatePicker) {
-            val datePickerState = rememberDatePickerState(
-                initialSelectedDateMillis = System.currentTimeMillis(),
-                selectableDates = PresentAndFutureSelectableDates()
-            )
-            DatePickerDialog(
-                onDismissRequest = { showDatePicker = false },
-                confirmButton = {
-                    TextButton(onClick = {
-                        datePickerState.selectedDateMillis?.let {
-                            selectedDateMillis = it
-                            fetchAvailableSlots(
-                                reservationManager,
-                                companyId,
-                                it
-                            ) { slots -> availableSlots = slots }
-                        }
-                        showDatePicker = false
-                    }) {
-                        Text(text = stringResource(R.string.confirm))
-                    }
+            ReservationCalendar(
```

Comment:
Revamped use of calendar picker to a reusable one. Amazing work. Reusability is very important!